app/assets/javascripts/Components/Result/result.jsx

```diff
@@ -541,6 +557,33 @@ class Result extends React.Component {
     });
   };
 
+  nextSubmission = (direction) => () => {
```

Comment:
I see what you're doing here but it's probably better not to use the nested lambdas like this (since we don't do it elsewhere in the code), and instead wrap the callback in a thunk below.